src/MessagingAttachment/MessagingAttachment.tsx

```diff
@@ -65,8 +67,11 @@ export const MessagingAttachment: React.FC<Props> = ({
         onClick={() => {
           if (isImageAttachment) {
             setAttachmentPreviewIsShowing(true);
+            if (onPreviewAttachment) {
+              onPreviewAttachment();
+            }
           } else {
-            onClickAttachment(attachmentID);
+            onClickDownload();
           }
         }}
```

Comment:
This looks great! Think it's much easier to reason about now, from the consumer side. 😄